### PR TITLE
Feat(eos_cli_config_gen): add various config options for sflow

### PR DESCRIPTION
**Pull request by ccsnw**
## Change Summary

Adding the following options for Sflow configuration:

a) SFlow source as IP under vrf and global
b) SFlow custom polling interval
c) SFlow BGP Extensions

Also reordered "sflow run" according to output in EOS cli

## Related Issue(s)

Fixes #<N/A>

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
a) and b) are self explanatory, just key: value

for c) I've decided to create an "enabled" and "disabled" extensions lists (more extensions might be added in the future?):

```
sflow:
  <snip>
  extensions:
    enabled:
      - <bgp | router | switch | tunnel >
    disabled:
      - <bgp | router | switch | tunnel >
```

could also use a map instead:

```
sflow:
  extensions:
    - name: < ext_name >
       enabled: < true | false >
```

Please let me know what fits better with the current overall design, or share a third proposal.

## How to test
Added testcases to molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
IMO option two of your proposals look more like the rest of AVD. It would also cover not having the same extension name twice, if you set `primary_key: name` in the schema for the list.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2 [hunk 1]
```diff
@@ -5,8 +5,8 @@
 ### SFlow Summary
 {%     if ((sflow.vrfs is arista.avd.defined) or (sflow.destinations is arista.avd.defined) or (sflow.source_interface is arista.avd.defined)) %}
 
-| VRF | SFlow Source Interface | SFlow Destination | Port |
-| --- | ---------------------- | ----------------- | ---- |
+| VRF | SFlow Source Interface or IP | SFlow Destination | Port |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2]:
```suggestion
| VRF | SFlow Source | SFlow Destination | Port |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/sflow.j2 [hunk 2]
```diff
@@ -28,15 +34,28 @@ sflow vrf {{ vrf.name }} source-interface {{ vrf.source_interface }}
 {%         endif %}
 {{ destination_cli }}
 {%     endfor %}
+{%     if sflow.source is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/sflow.j2]: Are source and source_interface allowed at the same time in CLI? If not, use an `elif` and choose one as the winner. (Document that if both are configured, xxx will take precedence)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2 (hunk 1 above)
Comment (ccsnw) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2]: makes sense!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/sflow.j2 (hunk 2 above)
Comment (ccsnw) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/sflow.j2]: you are right Claus, they are mutually exclusive

**Comment by ccsnw** [comment]
> IMO option two of your proposals look more like the rest of AVD. It would also cover not having the same extension name twice, if you set `primary_key: name` in the schema for the list.

changed accordingly, better complies with the rest of the code. thanks!

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/sflow.yml [hunk 3]
```diff
@@ -16,13 +16,32 @@ sflow:
...
           port: 333
       source_interface: Ethernet2
+    BBB:
+      destinations:
+        10.6.75.62:
+      source: 1.1.1.1
+    CCC:
+      destinations:
+        10.6.75.62:
+      source_interface: Management 1
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/sflow.yml]:
```suggestion
      source_interface: Management1
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 4]
```diff
@@ -2325,12 +2325,15 @@ trackers:
...
   sample: < sample_rate >
   dangerous: < true | false >
+  polling_interval: < polling_interval_seconds >
   vrfs:
     <vrf_name_1>:
       destinations:
         < sflow_destination_ip_1>:
         < sflow_destination_ip_2>:
           port: < port_number >
+      # source and source_interface are mutually exclusive. If both are defined, source_interfaces takes precedence
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
      # source and source_interface are mutually exclusive. If both are defined, source_interface takes precedence
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 5]
```diff
@@ -2340,7 +2343,12 @@ sflow:
     < sflow_destination_ip_1 >:
       port: < port_number >
     < sflow_destination_ip_2 >:
+  # source and source_interface are mutually exclusive. If both are defined, source_interfaces takes precedence
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
  # source and source_interface are mutually exclusive. If both are defined, source_interface takes precedence
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml [hunk 6]
```diff
@@ -60,9 +65,24 @@ keys:
               display_name: Port Number
               convert_types:
               - str
+      source:
+        type: str
+        display_name: Source IP
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml]:
```suggestion
        description: Source IP Address
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml [hunk 7]
```diff
@@ -41,6 +43,9 @@ keys:
                     display_name: Port Number
                     convert_types:
                     - str
+            source:
+              type: str
+              display_name: Source IP
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml]:
```suggestion
              description: Source IP Address
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2 [hunk 8]
```diff
@@ -65,6 +73,20 @@ sFlow hardware accelerated Sample Rate: {{ sflow.hardware_acceleration.sample }}
...
+### SFlow Extensions
+
+| Extension | Enabled |
+| --------- | ------- |
+{%         for extension in sflow.extensions | arista.avd.natural_sort('name') if extension.name is arista.avd.defined and extension.enabled is arista.avd.defined %}
+{%             if extension.enabled is arista.avd.defined(true) %}
+| {{ extension.name }} | True |
+{%             else %}
+| {{ extension.name }} | False |
+{%             endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/sflow.j2]:
```suggestion
| {{ extension.name }} | {{ extension.enabled }} |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml [hunk 9]
```diff
@@ -60,9 +65,24 @@ keys:
...
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Extension Name
+            enabled:
+              type: bool
+              description: Enable or Disable Extension
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml]: Your logic in the eos template only works if this is set. So either set enabled as required _or_ set a `default: true` and adjust the logic.
```suggestion
              description: Enable or Disable Extension
              required: true
```
FYI the primary key is always `required: true` implicitly, so it is up to you if you also want to add it there.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml (hunk 7 above)
Comment (ccsnw) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml]: I will also add "Address" to the descriptions of Destination.. to harmonize it

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml (hunk 9 above)
Comment (ccsnw) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml]: I'm setting "enabled" as required. I think the user should explicitly enable or disable these extensions

**Pull request closed without merging**